speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ConfirmationMenu.cpp

```diff
@@ -0,0 +1,80 @@
+#include <tgfclient.h>
+
+#include "legacymenu.h"
+#include "ConfigEnums.h"
+#include "mediator.h"
+#include "ConfirmationMenu.h"
+#include "mainmenu.h"
+
+static void* s_menuHandle = nullptr;
+
+/// @brief tells the mediator to not save experiment data and close SpeedDreams
+static void OnAcceptExit(void* /* dummy */)
+{
+    SMediator::GetInstance()->SetSaveRaceToDatabase(false);
+    LmRaceEngine().abortRace();  // Do cleanup to get back correct setup files
+    LegacyMenu::self().quit();
+}
+
+/// @brief tells the mediator to not save experiment data and restart SpeedDreams
+static void OnAcceptRestart(void* /* dummy */)
+{
+    SMediator::GetInstance()->SetSaveRaceToDatabase(false);
+    LmRaceEngine().restartRace();
+}
+
+/// @brief tells the mediator to save experiment data and abort the race
+static void OnAcceptAbort(void* /* dummy */)
+{
+    SMediator::GetInstance()->SetSaveRaceToDatabase(false);
+    LmRaceEngine().abortRace();
+}
+
+/// @brief                  create a confirmation screen
+/// @param p_prevMenu       the previous menu from where it came
+/// @param P_saveWayVersion enum that decided how you got to the save screen
+void* ConfirmationMenuInit(void* p_prevMenu, RaceEndType p_saveWayVersion)
```

Comment:
I think it's clearer to just have the param indicate that it contains the raceEndType, which you then USE to determine which save screen you go to.
```suggestion
void* ConfirmationMenuInit(void* p_prevMenu, RaceEndType p_raceEndType)
```